docs/configuration/http_conn_man/route_config/route.rst

```diff
@@ -40,15 +41,28 @@ path
 
 cluster
   *(sometimes required, string)* If the route is not a redirect (*host_redirect* and/or
-  *path_redirect* is not specified), one of *cluster* or *weighted_clusters* must be specified.
-  When a *cluster* is specified, its value indicates the upstream cluster to which the request
-  should be forwarded to.
+  *path_redirect* is not specified), one of *cluster*, *cluster_header*, or *weighted_clusters* must
+  be specified. When *cluster* is specified, its value indicates the upstream cluster to which the
+  request should be forwarded to.
+
+cluster_header
```

Comment:
Awesome :-) Just so im 100% clear, this is still going to work if the clusters are not known a-priori? (that is, they are discovered via CDS?)